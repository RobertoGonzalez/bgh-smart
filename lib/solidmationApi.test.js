const nock = require('nock')
const path = require('path')
const solidmation = require('./solidmationApi')

/* test('should fail constructor', () => {
  // const api = new solidmation()
  expect(new solidmation()).toThrowError('Missing credentials')
}) */

const fakeEmail = 'test@domain.com'
const fakePass = 'some#pass'
const fakeToken = '1A21423F-1EFB-498C-B1B7-0FFF8E6AFABB'

test('should login successfully', async () => {
  const req = nock('https://bgh-services.solidmation.com')
    .post('/control/LoginPage.aspx/DoStandardLogin', { user: fakeEmail, password: fakePass})
    .reply(200, { d: fakeToken })
  
  const api = new solidmation(fakeEmail, fakePass)

  const getToken = await api.login()
  expect(getToken).toBe(fakeToken)
  expect(api.auth.token).toBe(fakeToken)
})

test('should call req and retry login', async () => {
  nock('https://bgh-services.solidmation.com')
    .post('/control/LoginPage.aspx/DoStandardLogin', { user: fakeEmail, password: fakePass})
    .reply(200, { d: fakeToken })

  nock('https://bgh-services.solidmation.com')
    .post('/1.0/fake', { Token: fakeToken})
    .reply(200, {})

  const api = new solidmation(fakeEmail, fakePass)
  const doReq = await api.req('/fake', {})

  expect(doReq).toEqual({})
  expect(nock.isDone()).toBe(true)
})

describe('getHomes', () => {
  const api = new solidmation(fakeEmail, fakePass)
  api.setToken(fakeToken)

  beforeEach(() => {
    nock('https://bgh-services.solidmation.com')
      .post('/1.0/HomeCloudService.svc/EnumHomes')
      .replyWithFile(200, path.join(__dirname, '__responses__/EnumHomes.example.json'), {
        'Content-Type': 'application/json',
      })
  })

  test('should return an array with all homes', async () => {
    const homes = await api.  ()
    expect(nock.isDone()).toBe(true)
    expect(homes.length).toBe(2)
  })

  test('should return an array with filtered home', async () => {
    const homes = await api.getHomes(['DemoHome'])
    expect(nock.isDone()).toBe(true)
    expect(homes.length).toBe(1)
  })

  test('should return an array with one filtered home', async () => {
    const homes = await api.getHomes(['DemoHome', 'TestHome'])
    expect(nock.isDone()).toBe(true)
    expect(homes.length).toBe(1)
  })

  test('should match snapshot', async () => {
    const homes = await api.getHomes()
    expect(homes).toMatchSnapshot()
  })
  
})

describe('GetDataPacket', () => {
  const api = new solidmation(fakeEmail, fakePass)
  api.setToken(fakeToken)

  test('should match snapshot', async () => {
    nock('https://bgh-services.solidmation.com')
      .post('/1.0/HomeCloudService.svc/GetDataPacket')
      .replyWithFile(200, path.join(__dirname, '__responses__/GetDataPacket.json'), {
        'Content-Type': 'application/json',
      })
    const data = await api.getDataPacket(123)
    expect(data).toMatchSnapshot()
  })
})

describe('getDevicesForHomeId', () => {
  const api = new solidmation(fakeEmail, fakePass)
  api.setToken(fakeToken)

  test('should match snapshot', async () => {
    nock('https://bgh-services.solidmation.com')
      .post('/1.0/HomeCloudService.svc/GetDataPacket')
      .replyWithFile(200, path.join(__dirname, '__responses__/GetDataPacket.json'), {
        'Content-Type': 'application/json',
      })
    const data = await api.getDevicesForHomeId(123)
    expect(data).toMatchSnapshot()
  })
})

describe('getDevices', () => {
  const api = new solidmation(fakeEmail, fakePass)
  api.setToken(fakeToken)

  test('should match snapshot', async () => {
    nock('https://bgh-services.solidmation.com')
      .post('/1.0/HomeCloudService.svc/EnumHomes')
      .replyWithFile(200, path.join(__dirname, '__responses__/EnumHomes.example.json'), {
        'Content-Type': 'application/json',
      })

    nock('https://bgh-services.solidmation.com')
      .post('/1.0/HomeCloudService.svc/GetDataPacket')
      .twice()
      .replyWithFile(200, path.join(__dirname, '__responses__/GetDataPacket.json'), {
        'Content-Type': 'application/json',
      })
    
    await api.getHomes()
    const data = await api.getDevices()
    expect(data).toMatchSnapshot()
  })
})

describe('Endpoint Cache', () => {
  test('should return false', () => {
    const api = new solidmation(fakeEmail, fakePass)
    expect(api.hasCache(123)).toBe(false)
  })

  test('should return true', () => {
    const api = new solidmation(fakeEmail, fakePass)
    api.setCache(123)
    expect(api.hasCache(123)).toBe(true)
  })

  test('should return false', (done) => {
    const api = new solidmation(fakeEmail, fakePass)
    api.cacheMs = 50
    api.setCache(123)
    setTimeout(() => {
      expect(api.hasCache(123)).toBe(false)
      done()
    }, 100)
  })
})

describe('getDevicesForHomeId', () => {
  const api = new solidmation(fakeEmail, fakePass)
  api.setToken(fakeToken)

  test('should match snapshot', async () => {
    nock('https://bgh-services.solidmation.com')
      .post('/1.0/HomeCloudService.svc/EnumHomes')
      .replyWithFile(200, path.join(__dirname, '__responses__/EnumHomes.example.json'), {
        'Content-Type': 'application/json',
      })

    nock('https://bgh-services.solidmation.com')
      .post('/1.0/HomeCloudService.svc/GetDataPacket')
      .replyWithFile(200, path.join(__dirname, '__responses__/GetDataPacket.json'), {
        'Content-Type': 'application/json',
      })
    
    const data = await api.getDevicesForHomeId(123)
    expect(data).toMatchSnapshot()
  })
})
