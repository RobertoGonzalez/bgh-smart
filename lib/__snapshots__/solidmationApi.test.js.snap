// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetDataPacket should match snapshot 1`] = `
Object {
  "AccessControlInfo": Object {
    "EntryCode": "99999",
  },
  "Devices": Array [
    Object {
      "Address": "AC123",
      "AutomaticFirmwareUpdate": true,
      "Comments": null,
      "Description": "AC",
      "DeviceID": 60001,
      "DeviceModel": "HPA-4911",
      "DeviceModelID": 15,
      "FirmwareVersion": "1.0.0.17",
      "HomeID": 10005,
      "IsOnline": true,
      "NewFirmwareAvailable": false,
      "NumberOfEndpoints": 1,
      "Parameters": null,
      "PrivateKey": "XXXXXXXXY1",
      "ScriptObjectID": null,
      "TechnologyID": 5,
    },
  ],
  "EndpointValues": Array [
    Object {
      "EndpointID": 70001,
      "Values": Array [
        Object {
          "Value": "22",
          "ValueType": 13,
        },
        Object {
          "Value": "0",
          "ValueType": 14,
        },
        Object {
          "Value": "3",
          "ValueType": 15,
        },
        Object {
          "Value": "0",
          "ValueType": 16,
        },
        Object {
          "Value": "0",
          "ValueType": 17,
        },
        Object {
          "Value": "0",
          "ValueType": 18,
        },
        Object {
          "Value": "24",
          "ValueType": 20,
        },
        Object {
          "Value": "80",
          "ValueType": 59,
        },
        Object {
          "Value": "3909",
          "ValueType": 60,
        },
        Object {
          "Value": "71",
          "ValueType": 61,
        },
        Object {
          "Value": "0",
          "ValueType": 62,
        },
      ],
    },
  ],
  "Endpoints": Array [
    Object {
      "Address": "1",
      "Capabilities": 2555264,
      "Description": "AC",
      "DeviceAddress": "AC123",
      "DeviceID": 60001,
      "DeviceModelDesc": "HPA-4911",
      "EndpointID": 70001,
      "EndpointModelDesc": null,
      "EndpointType": 12,
      "Functional": true,
      "Groups": Array [
        Object {
          "Description": "Living",
          "GroupID": 20001,
          "HomeID": 10005,
        },
      ],
      "HomeID": 10005,
      "IsEditable": true,
      "MeteringUnitID": 0,
      "Parameters": Array [
        Object {
          "Name": "ConfiguredSensors",
          "Value": "0",
        },
        Object {
          "Name": "IRProtocolCode",
          "Value": "30",
        },
        Object {
          "Name": "RoomTemperatureOffset",
          "Value": "-400",
        },
        Object {
          "Name": "SetpointMaxC",
          "Value": "30",
        },
        Object {
          "Name": "SetpointMinC",
          "Value": "17",
        },
      ],
      "ScriptObjectID": null,
    },
  ],
  "Groups": Array [
    Object {
      "Description": "Living",
      "GroupID": 20001,
      "HomeID": 10005,
    },
  ],
  "Home": Object {
    "Description": "Home",
    "HomeID": 10005,
    "InvitedBy": "",
    "IsDefault": true,
    "IsOnline": true,
    "IsOwner": true,
    "Location": Object {
      "CityName": null,
      "CountryName": null,
      "Latitude": null,
      "Longitude": null,
      "Radius": 500,
      "StateName": null,
    },
    "RemoteAccessKey": "XXXXXXXXY2",
    "TimeZoneID": 1,
    "TimeZoneOffset": -180,
    "Virtual": true,
    "Weather": Object {
      "Temperature": null,
    },
  },
  "Macros": null,
  "NewSerials": Object {
    "Alarms": 0,
    "Devices": 1803500375,
    "EndpointValues": 1027422332,
    "Endpoints": 1561243994,
    "Groups": 1766414354,
    "Home": 1480175878,
    "Macros": 487638952,
    "Notifications": 487638952,
    "Scenes": 218289185,
  },
  "ResponseStatus": Object {
    "Messages": Array [],
    "Status": 0,
  },
  "Scenes": Array [
    Object {
      "Description": "A",
      "EndpointValues": Array [
        Object {
          "EndpointID": 70001,
          "Values": Array [
            Object {
              "Value": "0",
              "ValueType": 14,
            },
            Object {
              "Value": "255",
              "ValueType": 15,
            },
            Object {
              "Value": "-327.68",
              "ValueType": 20,
            },
          ],
        },
      ],
      "Geofence": Object {
        "OnEnter": false,
        "OnExit": false,
        "RequiresConfirmation": false,
      },
      "Groups": Array [
        Object {
          "Description": "Living",
          "GroupID": 20001,
          "HomeID": 10005,
        },
      ],
      "HomeID": 10005,
      "SceneID": 8000,
      "Triggers": Array [
        Object {
          "Description": "at 04:00 hs everyday",
          "EndDate": Object {
            "Day": 0,
            "Month": 0,
            "Year": 0,
          },
          "Frequency": 0,
          "HomeID": 10005,
          "IntervalMinutes": 0,
          "MonthDayNumber": 0,
          "MonthWeek": 0,
          "MonthWeekDay": 0,
          "OffsetMinutes": 0,
          "OnApril": false,
          "OnAugust": false,
          "OnDecember": false,
          "OnFebruary": false,
          "OnFriday": true,
          "OnJanuary": false,
          "OnJuly": false,
          "OnJune": false,
          "OnMarch": false,
          "OnMay": false,
          "OnMonday": true,
          "OnNovember": false,
          "OnOctober": false,
          "OnSaturday": true,
          "OnSeptember": false,
          "OnSunday": true,
          "OnThursday": true,
          "OnTuesday": true,
          "OnWednesday": true,
          "RepeatForMinutes": 0,
          "RepeatUntilTime": Object {
            "Hour": 0,
            "Millisecond": 0,
            "Minute": 0,
            "Second": 0,
          },
          "RepetitionsLeft": 0,
          "RunningTime": Object {
            "Hour": 4,
            "Millisecond": 0,
            "Minute": 0,
            "Second": 0,
          },
          "StartDate": Object {
            "Day": 0,
            "Month": 0,
            "Year": 0,
          },
          "TimeType": 3,
          "TriggerID": 4001,
          "TriggerSourceID": 0,
          "TriggerType": 2,
          "TriggersExecutionID": 0,
        },
      ],
      "TriggersAreEnabled": false,
    },
    Object {
      "Description": "B",
      "EndpointValues": Array [
        Object {
          "EndpointID": 70001,
          "Values": Array [
            Object {
              "Value": "0",
              "ValueType": 14,
            },
            Object {
              "Value": "255",
              "ValueType": 15,
            },
            Object {
              "Value": "-327.68",
              "ValueType": 20,
            },
          ],
        },
      ],
      "Geofence": Object {
        "OnEnter": false,
        "OnExit": false,
        "RequiresConfirmation": false,
      },
      "Groups": Array [
        Object {
          "Description": "Living",
          "GroupID": 20001,
          "HomeID": 10005,
        },
      ],
      "HomeID": 10005,
      "SceneID": 70001,
      "Triggers": Array [
        Object {
          "Description": "at 08:30 hs everyday",
          "EndDate": Object {
            "Day": 0,
            "Month": 0,
            "Year": 0,
          },
          "Frequency": 0,
          "HomeID": 10005,
          "IntervalMinutes": 0,
          "MonthDayNumber": 0,
          "MonthWeek": 0,
          "MonthWeekDay": 0,
          "OffsetMinutes": 0,
          "OnApril": false,
          "OnAugust": false,
          "OnDecember": false,
          "OnFebruary": false,
          "OnFriday": true,
          "OnJanuary": false,
          "OnJuly": false,
          "OnJune": false,
          "OnMarch": false,
          "OnMay": false,
          "OnMonday": true,
          "OnNovember": false,
          "OnOctober": false,
          "OnSaturday": true,
          "OnSeptember": false,
          "OnSunday": true,
          "OnThursday": true,
          "OnTuesday": true,
          "OnWednesday": true,
          "RepeatForMinutes": 0,
          "RepeatUntilTime": Object {
            "Hour": 0,
            "Millisecond": 0,
            "Minute": 0,
            "Second": 0,
          },
          "RepetitionsLeft": 0,
          "RunningTime": Object {
            "Hour": 8,
            "Millisecond": 0,
            "Minute": 30,
            "Second": 0,
          },
          "StartDate": Object {
            "Day": 0,
            "Month": 0,
            "Year": 0,
          },
          "TimeType": 3,
          "TriggerID": 4002,
          "TriggerSourceID": 0,
          "TriggerType": 2,
          "TriggersExecutionID": 0,
        },
      ],
      "TriggersAreEnabled": false,
    },
  ],
}
`;

exports[`getDevices should match snapshot 1`] = `
Array [
  Object {
    "Address": "1",
    "Capabilities": 2555264,
    "Description": "AC",
    "DeviceAddress": "AC123",
    "DeviceID": 60001,
    "DeviceModelDesc": "HPA-4911",
    "EndpointID": 70001,
    "EndpointModelDesc": null,
    "EndpointType": 12,
    "Functional": true,
    "Groups": Array [
      Object {
        "Description": "Living",
        "GroupID": 20001,
        "HomeID": 10005,
      },
    ],
    "HomeID": 10005,
    "IsEditable": true,
    "MeteringUnitID": 0,
    "Parameters": Array [
      Object {
        "Name": "ConfiguredSensors",
        "Value": "0",
      },
      Object {
        "Name": "IRProtocolCode",
        "Value": "30",
      },
      Object {
        "Name": "RoomTemperatureOffset",
        "Value": "-400",
      },
      Object {
        "Name": "SetpointMaxC",
        "Value": "30",
      },
      Object {
        "Name": "SetpointMinC",
        "Value": "17",
      },
    ],
    "ScriptObjectID": null,
    "device": Object {
      "Address": "AC123",
      "AutomaticFirmwareUpdate": true,
      "Comments": null,
      "Description": "AC",
      "DeviceID": 60001,
      "DeviceModel": "HPA-4911",
      "DeviceModelID": 15,
      "FirmwareVersion": "1.0.0.17",
      "HomeID": 10005,
      "IsOnline": true,
      "NewFirmwareAvailable": false,
      "NumberOfEndpoints": 1,
      "Parameters": null,
      "PrivateKey": "XXXXXXXXY1",
      "ScriptObjectID": null,
      "TechnologyID": 5,
    },
    "endpointParameters": Object {
      "maxTemp": "30",
      "minTemp": "17",
    },
    "endpointValues": Object {
      "currentTemp": "22",
      "desiredTempC": "24",
      "fanMode": "3",
      "mode": "0",
      "swingMode": 0,
    },
    "rawEndpointValues": Object {
      "EndpointID": 70001,
      "Values": Array [
        Object {
          "Value": "22",
          "ValueType": 13,
        },
        Object {
          "Value": "0",
          "ValueType": 14,
        },
        Object {
          "Value": "3",
          "ValueType": 15,
        },
        Object {
          "Value": "0",
          "ValueType": 16,
        },
        Object {
          "Value": "0",
          "ValueType": 17,
        },
        Object {
          "Value": "0",
          "ValueType": 18,
        },
        Object {
          "Value": "24",
          "ValueType": 20,
        },
        Object {
          "Value": "80",
          "ValueType": 59,
        },
        Object {
          "Value": "3909",
          "ValueType": 60,
        },
        Object {
          "Value": "71",
          "ValueType": 61,
        },
        Object {
          "Value": "0",
          "ValueType": 62,
        },
      ],
    },
  },
  Object {
    "Address": "1",
    "Capabilities": 2555264,
    "Description": "AC",
    "DeviceAddress": "AC123",
    "DeviceID": 60001,
    "DeviceModelDesc": "HPA-4911",
    "EndpointID": 70001,
    "EndpointModelDesc": null,
    "EndpointType": 12,
    "Functional": true,
    "Groups": Array [
      Object {
        "Description": "Living",
        "GroupID": 20001,
        "HomeID": 10005,
      },
    ],
    "HomeID": 10005,
    "IsEditable": true,
    "MeteringUnitID": 0,
    "Parameters": Array [
      Object {
        "Name": "ConfiguredSensors",
        "Value": "0",
      },
      Object {
        "Name": "IRProtocolCode",
        "Value": "30",
      },
      Object {
        "Name": "RoomTemperatureOffset",
        "Value": "-400",
      },
      Object {
        "Name": "SetpointMaxC",
        "Value": "30",
      },
      Object {
        "Name": "SetpointMinC",
        "Value": "17",
      },
    ],
    "ScriptObjectID": null,
    "device": Object {
      "Address": "AC123",
      "AutomaticFirmwareUpdate": true,
      "Comments": null,
      "Description": "AC",
      "DeviceID": 60001,
      "DeviceModel": "HPA-4911",
      "DeviceModelID": 15,
      "FirmwareVersion": "1.0.0.17",
      "HomeID": 10005,
      "IsOnline": true,
      "NewFirmwareAvailable": false,
      "NumberOfEndpoints": 1,
      "Parameters": null,
      "PrivateKey": "XXXXXXXXY1",
      "ScriptObjectID": null,
      "TechnologyID": 5,
    },
    "endpointParameters": Object {
      "maxTemp": "30",
      "minTemp": "17",
    },
    "endpointValues": Object {
      "currentTemp": "22",
      "desiredTempC": "24",
      "fanMode": "3",
      "mode": "0",
      "swingMode": 0,
    },
    "rawEndpointValues": Object {
      "EndpointID": 70001,
      "Values": Array [
        Object {
          "Value": "22",
          "ValueType": 13,
        },
        Object {
          "Value": "0",
          "ValueType": 14,
        },
        Object {
          "Value": "3",
          "ValueType": 15,
        },
        Object {
          "Value": "0",
          "ValueType": 16,
        },
        Object {
          "Value": "0",
          "ValueType": 17,
        },
        Object {
          "Value": "0",
          "ValueType": 18,
        },
        Object {
          "Value": "24",
          "ValueType": 20,
        },
        Object {
          "Value": "80",
          "ValueType": 59,
        },
        Object {
          "Value": "3909",
          "ValueType": 60,
        },
        Object {
          "Value": "71",
          "ValueType": 61,
        },
        Object {
          "Value": "0",
          "ValueType": 62,
        },
      ],
    },
  },
]
`;

exports[`getDevicesForHomeId should match snapshot 1`] = `
Array [
  Object {
    "Address": "1",
    "Capabilities": 2555264,
    "Description": "AC",
    "DeviceAddress": "AC123",
    "DeviceID": 60001,
    "DeviceModelDesc": "HPA-4911",
    "EndpointID": 70001,
    "EndpointModelDesc": null,
    "EndpointType": 12,
    "Functional": true,
    "Groups": Array [
      Object {
        "Description": "Living",
        "GroupID": 20001,
        "HomeID": 10005,
      },
    ],
    "HomeID": 10005,
    "IsEditable": true,
    "MeteringUnitID": 0,
    "Parameters": Array [
      Object {
        "Name": "ConfiguredSensors",
        "Value": "0",
      },
      Object {
        "Name": "IRProtocolCode",
        "Value": "30",
      },
      Object {
        "Name": "RoomTemperatureOffset",
        "Value": "-400",
      },
      Object {
        "Name": "SetpointMaxC",
        "Value": "30",
      },
      Object {
        "Name": "SetpointMinC",
        "Value": "17",
      },
    ],
    "ScriptObjectID": null,
    "device": Object {
      "Address": "AC123",
      "AutomaticFirmwareUpdate": true,
      "Comments": null,
      "Description": "AC",
      "DeviceID": 60001,
      "DeviceModel": "HPA-4911",
      "DeviceModelID": 15,
      "FirmwareVersion": "1.0.0.17",
      "HomeID": 10005,
      "IsOnline": true,
      "NewFirmwareAvailable": false,
      "NumberOfEndpoints": 1,
      "Parameters": null,
      "PrivateKey": "XXXXXXXXY1",
      "ScriptObjectID": null,
      "TechnologyID": 5,
    },
    "endpointParameters": Object {
      "maxTemp": "30",
      "minTemp": "17",
    },
    "endpointValues": Object {
      "currentTemp": "22",
      "desiredTempC": "24",
      "fanMode": "3",
      "mode": "0",
      "swingMode": 0,
    },
    "rawEndpointValues": Object {
      "EndpointID": 70001,
      "Values": Array [
        Object {
          "Value": "22",
          "ValueType": 13,
        },
        Object {
          "Value": "0",
          "ValueType": 14,
        },
        Object {
          "Value": "3",
          "ValueType": 15,
        },
        Object {
          "Value": "0",
          "ValueType": 16,
        },
        Object {
          "Value": "0",
          "ValueType": 17,
        },
        Object {
          "Value": "0",
          "ValueType": 18,
        },
        Object {
          "Value": "24",
          "ValueType": 20,
        },
        Object {
          "Value": "80",
          "ValueType": 59,
        },
        Object {
          "Value": "3909",
          "ValueType": 60,
        },
        Object {
          "Value": "71",
          "ValueType": 61,
        },
        Object {
          "Value": "0",
          "ValueType": 62,
        },
      ],
    },
  },
]
`;

exports[`getDevicesForHomeId should match snapshot 2`] = `
Array [
  Object {
    "Address": "1",
    "Capabilities": 2555264,
    "Description": "AC",
    "DeviceAddress": "AC123",
    "DeviceID": 60001,
    "DeviceModelDesc": "HPA-4911",
    "EndpointID": 70001,
    "EndpointModelDesc": null,
    "EndpointType": 12,
    "Functional": true,
    "Groups": Array [
      Object {
        "Description": "Living",
        "GroupID": 20001,
        "HomeID": 10005,
      },
    ],
    "HomeID": 10005,
    "IsEditable": true,
    "MeteringUnitID": 0,
    "Parameters": Array [
      Object {
        "Name": "ConfiguredSensors",
        "Value": "0",
      },
      Object {
        "Name": "IRProtocolCode",
        "Value": "30",
      },
      Object {
        "Name": "RoomTemperatureOffset",
        "Value": "-400",
      },
      Object {
        "Name": "SetpointMaxC",
        "Value": "30",
      },
      Object {
        "Name": "SetpointMinC",
        "Value": "17",
      },
    ],
    "ScriptObjectID": null,
    "device": Object {
      "Address": "AC123",
      "AutomaticFirmwareUpdate": true,
      "Comments": null,
      "Description": "AC",
      "DeviceID": 60001,
      "DeviceModel": "HPA-4911",
      "DeviceModelID": 15,
      "FirmwareVersion": "1.0.0.17",
      "HomeID": 10005,
      "IsOnline": true,
      "NewFirmwareAvailable": false,
      "NumberOfEndpoints": 1,
      "Parameters": null,
      "PrivateKey": "XXXXXXXXY1",
      "ScriptObjectID": null,
      "TechnologyID": 5,
    },
    "endpointParameters": Object {
      "maxTemp": "30",
      "minTemp": "17",
    },
    "endpointValues": Object {
      "currentTemp": "22",
      "desiredTempC": "24",
      "fanMode": "3",
      "mode": "0",
      "swingMode": 0,
    },
    "rawEndpointValues": Object {
      "EndpointID": 70001,
      "Values": Array [
        Object {
          "Value": "22",
          "ValueType": 13,
        },
        Object {
          "Value": "0",
          "ValueType": 14,
        },
        Object {
          "Value": "3",
          "ValueType": 15,
        },
        Object {
          "Value": "0",
          "ValueType": 16,
        },
        Object {
          "Value": "0",
          "ValueType": 17,
        },
        Object {
          "Value": "0",
          "ValueType": 18,
        },
        Object {
          "Value": "24",
          "ValueType": 20,
        },
        Object {
          "Value": "80",
          "ValueType": 59,
        },
        Object {
          "Value": "3909",
          "ValueType": 60,
        },
        Object {
          "Value": "71",
          "ValueType": 61,
        },
        Object {
          "Value": "0",
          "ValueType": 62,
        },
      ],
    },
  },
]
`;

exports[`getHomes should match snapshot 1`] = `
Array [
  Object {
    "Description": "DemoHome",
    "HomeID": 10005,
    "InvitedBy": null,
    "IsDefault": false,
    "IsOnline": true,
    "IsOwner": true,
    "Location": Object {
      "CityName": null,
      "CountryName": null,
      "Latitude": null,
      "Longitude": null,
      "Radius": 500,
      "StateName": null,
    },
    "RemoteAccessKey": "XXXXXXXX01",
    "TimeZoneID": 1,
    "TimeZoneOffset": -180,
    "Virtual": true,
    "Weather": Object {
      "Temperature": null,
    },
  },
  Object {
    "Description": "Home",
    "HomeID": 10006,
    "InvitedBy": null,
    "IsDefault": true,
    "IsOnline": true,
    "IsOwner": true,
    "Location": Object {
      "CityName": null,
      "CountryName": null,
      "Latitude": null,
      "Longitude": null,
      "Radius": 500,
      "StateName": null,
    },
    "RemoteAccessKey": "XXXXXXXX02",
    "TimeZoneID": 1,
    "TimeZoneOffset": -180,
    "Virtual": true,
    "Weather": Object {
      "Temperature": null,
    },
  },
]
`;
